src/js/components/Tabs/Tabs.js

```diff
@@ -103,19 +238,179 @@ const Tabs = forwardRef(
         {...rest}
         background={theme.tabs.background}
       >
-        <StyledTabsHeader
-          as={Box}
-          direction="row"
-          justify={justify}
-          alignSelf={alignControls}
-          flex={false}
-          wrap
-          background={theme.tabs.header.background}
-          gap={theme.tabs.gap}
-          {...tabsHeaderStyles}
-        >
-          {tabs}
-        </StyledTabsHeader>
+        <Box direction={overflow ? 'row' : 'column'} {...tabsHeaderStyles}>
+          {overflow && (
+            <Button
+              a11yTitle="Previous Tab"
+              disabled={disableLeftArrow}
+              // removed from tabIndex, button is redundant for keyboard users
+              tabIndex={-1}
+              onClick={() => {
+                let scrolledToIndex;
```

Comment:
There seems to be a certain amount of overlap between this code and `moveIntoView()`.  Would it be possible to combine them somehow?

It seems a little odd to me here that we're looping through all of the tabs here. Can we do something a simple as "if the left or right edge of the Tab we want to be visible isn't, scroll so it is by the amount that it isn't?

Reply:
Consolidated code in latest commit